src/libraries/System.Private.CoreLib/src/System/Reflection/CustomAttributeData.cs

```diff
@@ -51,22 +55,26 @@ public override string ToString()
             vsb.Append(Constructor.DeclaringType!.FullName);
             vsb.Append('(');
 
-            bool first = true;
-
-            int count = ConstructorArguments.Count;
+            IList<CustomAttributeTypedArgument> constructorArguments = ConstructorArguments;
+            int count = constructorArguments.Count;
             for (int i = 0; i < count; i++)
             {
-                if (!first) vsb.Append(", ");
-                vsb.Append(ConstructorArguments[i].ToString());
-                first = false;
+                if (i != 0)
+                {
+                    vsb.Append(", ");
+                }
+                vsb.Append(constructorArguments[i].ToString());
```

Comment:
Allocating a string to pass to Append is a bit messy.

Reply:
It is how it was before this change too.

ValueStringBuilder does not have overload that takes object to make ToString explicit, so there is no easy way around it.